pkg/nfd-master/nfd-master.go

```diff
@@ -972,18 +979,32 @@ func (m *nfdMaster) updateNodeObject(cli *kubernetes.Clientset, nodeName string,
 			extendedResourceKeys = append(extendedResourceKeys, strings.TrimPrefix(key, nfdv1alpha1.FeatureLabelNs+"/"))
 		}
 		sort.Strings(extendedResourceKeys)
-		annotations[m.instanceAnnotation(nfdv1alpha1.ExtendedResourceAnnotation)] = strings.Join(extendedResourceKeys, ",")
+		nfdAnnotations[m.instanceAnnotation(nfdv1alpha1.ExtendedResourceAnnotation)] = strings.Join(extendedResourceKeys, ",")
+	}
+
+	// Store names of feature annotations in an annotation
+	annotationKeys := make([]string, 0, len(featureAnnotations))
```

Comment:
Suggest to only set this annotation if some feature annotations are present

```suggestion
if len(featureAnnotations) > 0 { 
	annotationKeys := make([]string, 0, len(featureAnnotations))
```